src/librbd/deep_copy/ObjectCopyRequest.cc

```diff
@@ -338,6 +338,20 @@ void ObjectCopyRequest<I>::send_update_object_map() {
     return;
   }
 
+  auto &dst_object_state = *m_dst_object_state.begin();
+  auto it = m_snap_map.find(dst_object_state.first);
+  assert(it != m_snap_map.end());
+  auto dst_snap_id = it->second.front();
+  auto object_state = dst_object_state.second;
+  m_dst_object_state.erase(m_dst_object_state.begin());
+
+  if (dst_snap_id == CEPH_NOSNAP &&
+      Striper::get_num_objects(m_dst_image_ctx->layout,
```

Comment:
Not holding any locks here -- while the image shouldn't be updated behind the scenes since it's a new image, probably safer to grab the locks. Alternatively, perhaps this [1] should be tweaked to also test if the start position is beyond the bounds.

[1] https://github.com/ceph/ceph/blob/master/src/librbd/ObjectMap.cc#L305

Reply:
@dillaman Thanks. I tweaked the object map update but in a different place, where it actually failed.